src/sentry/lang/javascript/processor.py

```diff
@@ -331,7 +334,14 @@ def fetch_file(url, project=None, release=None, dist=None, allow_scraping=True):
         with metrics.timer('sourcemaps.fetch'):
             result = http.fetch_file(url, headers=headers, verify_ssl=verify_ssl)
             z_body = zlib.compress(result.body)
-            cache.set(cache_key, (url, result.headers, z_body, result.status, result.encoding), 60)
+            cache.set(
+                cache_key,
+                (url,
+                 result.headers,
+                 z_body,
+                 result.status,
+                 result.encoding),
+                get_max_age(headers))
```

Comment:
You're checking the wrong `headers` here. You need to be looking at `result.headers`. `headers` is the _request_ headers, not the `response`.